core/jazz_slack-event-handler/components/utils.js

```diff
@@ -0,0 +1,143 @@
+// =========================================================================
+// Copyright © 2017 T-Mobile USA, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// =========================================================================
+
+const format = require("string-template");
+
+function getNotificationMessage(serviceDetails, payload, configData) {
+  var serviceContxt = JSON.parse(payload.SERVICE_CONTEXT.S);
+  var params = {
+    'service_name': serviceDetails.service,
+    'domain_name': serviceDetails.domain,
+    'environment_name': serviceContxt.environment,
+    'event_type': payload.EVENT_TYPE.S,
+    'event_name': (payload.EVENT_NAME.S).toLowerCase(),
+    'build_status': payload.EVENT_STATUS.S,
+    'event_status': payload.EVENT_STATUS.S,
+    'repo_url': serviceContxt.repository,
+    'overview_url': configData.SERVICE_LINK + serviceDetails.id,
+    'jenkins_url': serviceContxt.provider_build_url,
+    'endpoint_url': serviceContxt.endpoint_url,
+    'notifications': configData.NOTIFICATION_MESSAGE,
+    'error': serviceContxt.error
+  };
+
+  var slackNotification = {
+    "pretext": "",
+    "text": "",
+    "color": "",
+  };
+
+  var pretext, text;
+  if (params.service_name && params.domain_name && params.event_type) {
```

Comment:
I had a comment on this that there is too much logic here and needs to be extracted. Can you please tell me why was this merged without addressing the comment?